pilot/pkg/model/push_context.go

```diff
@@ -226,8 +226,30 @@ type PushRequest struct {
 	// Start represents the time a push was started. This represents the time of adding to the PushQueue.
 	// Note that this does not include time spent debouncing.
 	Start time.Time
+
+	// Reason represents the reason for requesting a push. This should only be a fixed set of values,
+	// to avoid unbounded cardinality in metrics. If this is not set, it may be automatically filled in later.
+	// There may be multiple reasons, only if there are multiple requests merge together
+	Reason []TriggerReason
 }
 
+type TriggerReason string
+
+var (
+	// Describes a push triggered by an Endpoint change
+	EndpointUpdate TriggerReason = "endpoint"
+	// Describes a push triggered by a config change
+	ConfigUpdate TriggerReason = "config"
+	// Describes a push triggered by a Service change
+	ServiceUpdate TriggerReason = "service"
+	// Describes a push triggered by a change to an individual proxy (such as label change)
+	ProxyUpdate TriggerReason = "proxy"
+	// Describes a push triggered by a change to bootstrap config, such as mesh config
+	BootstrapUpdate TriggerReason = "bootstrap"
```

Comment:
BootStrapUpdate seems to indicate it is loaded once - May be "GlobalMeshConfig"  or some thing similar?

Reply:
Its not just mesh config, its about other global state changes. bootstrap isn't a good name, but I think mesh config isn't either... not sure a good one, I am bad at names

Reply:
yeah. How about "GlobalConfig" / "GlobalEnvUpdate"?